Sources/Classes/RSClient.m

```diff
@@ -541,4 +541,37 @@ - (void) onIntegrationReady:(id<RSIntegrationFactory>)factory withCallback:(Call
     }
 }
 
+#pragma mark - Deep Link Track
+
+- (void)openURL:(NSURL *)url options:(NSDictionary *)options
+{
+    if ([RSClient getOptStatus]) {
+            [self reportDiscardedEvent];
+            return;
+    }
+    NSString *urlString = url.absoluteString;
+    NSMutableDictionary *properties = [NSMutableDictionary dictionaryWithCapacity:options.count + 2];
+    NSArray *paramArray = [RSUtils extractParamFromURL:url];
+    if (paramArray.count > 0) {
+        // Iterate through the query items
+        for (NSURLQueryItem *item in paramArray) {
+            [RSLogger logVerbose:[[NSString alloc] initWithFormat:@"Parameter name: %@, value: %@", item.name, item.value]];
+            properties[item.name] = item.value;
+        }
+    }
+    properties[@"url"] = urlString;
+    if (options != nil) {
+            for (NSString *key in options) {
+                properties[@"options"] = options;
+            }
+        }
+    [self track:@"Deep Link Opened" properties:[properties copy]];
+}
+
+- (void)openURL:(NSURL *)url
+{
+    NSDictionary *options = [[NSDictionary alloc] init];
+    [self openURL:url options:options];
+}
```

Comment:
In this line `properties[@"options"] = options;`, we are creating another `options` field; instead we can copy the content directly into the properties object. 

Please use the following code snippet. Over here, I've optimised the code and fixed indentation:

```suggestion
- (void)openURL:(NSURL *)url options:(NSDictionary<NSString *,NSObject *> *)options {
    if ([RSClient getOptStatus]) {
        [self reportDiscardedEvent];
        return;
    }
    NSString *urlString = url.absoluteString;
    NSMutableDictionary<NSString *,NSObject *> *properties = [[NSMutableDictionary alloc] init];
    NSArray *paramArray = [RSUtils extractParamFromURL:url];
    if (paramArray.count > 0) {
        // Iterate through the query items
        for (NSURLQueryItem *item in paramArray) {
            [RSLogger logVerbose:[[NSString alloc] initWithFormat:@"Parameter name: %@, value: %@", item.name, item.value]];
            properties[item.name] = item.value;
        }
    }
    if (options != nil) {
        for (NSString *key in options) {
            properties[key] = options[key];
        }
    }
    properties[@"url"] = urlString;
    [self track:@"Deep Link Opened" properties:[properties copy]];
}

- (void)openURL:(NSURL *)url {
    [self openURL:url options:nil];
}
```